packages/yourloops/pages/hcp/patients/table.tsx

```diff
@@ -155,55 +161,64 @@ function PatientRow(props: PatientElementProps): JSX.Element {
-          </Tooltip>
-        </TableCell>
-        <TableCell id={`${rowId}-lastname`}>{lastName}</TableCell>
-        <TableCell id={`${rowId}-firstname`}>{firstName}</TableCell>
-        <TableCell id={`${rowId}-tir`}>{tir}</TableCell>
-        <TableCell id={`${rowId}-tbr`}>{tbr}</TableCell>
-        <TableCell id={`${rowId}-upload`}>{lastUpload}</TableCell>
-      </TableRow>
-    );
-  }
+  const firstRowIcon = isPendingInvitation ?
+    (<Tooltip
+      id={`${rowId}-tooltip-pending`}
+      title={t("pending-invitation") as string}
+      aria-label={t("pending-invitation")}
+      placement="bottom"
+    >
+      <AccessTimeIcon id={`${rowId}-pendingicon`} />
+    </Tooltip>) :
+    (<IconActionButton
+      icon={isFlagged ? <FlagIcon id={`${rowId}-flagged`} /> : <FlagOutlineIcon id={`${rowId}-un-flagged`} />}
+      id={`${rowId}-icon-button-flag`}
+      onClick={onClickFlag}
+      className={`${classes.flag} patient-flag-button`}
+    />);
 
   return (
-    <TableRow id={rowId} tabIndex={-1} hover onClick={onRowClick} className={`${classes.tableRow} patients-list-row`} data-userid={userId} data-email={email} ref={rowRef}>
-      <TableCell id={`${rowId}-icon`}>
-        <IconButton id={`${rowId}-icon-button-flag`} className={classes.flag} aria-label={t("aria-flag-patient")} size="small" onClick={onClickFlag}>
-          {isFlagged ? <FlagIcon id={`${rowId}-flagged`} /> : <FlagOutlineIcon id={`${rowId}-un-flagged`} />}
-        </IconButton>
-      </TableCell>
+    <TableRow
+      id={rowId}
+      tabIndex={-1}
+      hover
+      onClick={isPendingInvitation ? undefined : onRowClick}
+      className={`${classes.tableRow} patients-list-row`}
+      data-userid={userId}
+      data-email={email}
+      ref={rowRef}
+    >
+      <TableCell id={`${rowId}-icon`}>{firstRowIcon}</TableCell>
       <TableCell id={`${rowId}-lastname`}>{lastName}</TableCell>
       <TableCell id={`${rowId}-firstname`}>{firstName}</TableCell>
       <TableCell id={`${rowId}-tir`}>{tir}</TableCell>
       <TableCell id={`${rowId}-tbr`}>{tbr}</TableCell>
       <TableCell id={`${rowId}-upload`}>{lastUpload}</TableCell>
+      <TableCell id={`${rowId}-remove-icon`}>
+        <IconActionButton
+          icon={<PersonRemoveIcon />}
+          className="remove-patient-hcp-view-button"
+          onClick={onClickRemoveIcon}
+        />
+      </TableCell>
     </TableRow>
   );
 }
 
 function PatientListTable(props: PatientListProps): JSX.Element {
-  const { patients, flagged, order, orderBy, onClickPatient, onFlagPatient, onSortList } = props;
+  const { patients, flagged, order, orderBy, onClickPatient, onFlagPatient, onSortList, onClickRemovePatient } = props;
   const { t } = useTranslation("yourloops");
   const classes = patientListStyle();
-
-  const trNA = t("N/A");
-
   const patientsRows = patients.map(
     (patient: TeamUser): JSX.Element => (
       <PatientRow
         key={patient.userid}
-        trNA={trNA}
+        trNA={t("N/A")}
```

Comment:
The purpose of trNA was to avoid doing this translation for every rows, up to 3x per row.